src/transformers/pipelines.py

```diff
@@ -1664,6 +1665,270 @@ def __call__(
             return results
 
 
+class Conversation:
```

Comment:
This abstraction seems a bit weird to me because we are mutating so much of its properties from the pipeline below. i.e. it's not very self-contained. I'm wondering if we should move some of the mutating methods to the class directly.

Reply:
I created some class functions to avoid direct field mutation - hopefully it goes in the right direction